### PR TITLE
add cluster.local

### PR DESCRIPTION
**Pull request by mangirdaz**
I run this in env where they use dnsmasq and intermediate dns. 
It fails as it needs the full path to check it. It should not change default behavior :)

### CODE REVIEW

**Comment by sdodson** [comment]
If `cluster.local` isn't in the host's dns search path then other components will break too. Why isn't it added by the dispatcher script?

**Comment by mangirdaz** [comment]
resolv.conf: 
```
-bash-4.2# cat /etc/resolv.conf 
# nameserver updated by /etc/NetworkManager/dispatcher.d/99-origin-dns.sh
# Generated by NetworkManager
search ocpcluster.local
nameserver 192.168.20.160
```

dnsmasq:
```
-bash-4.2# systemctl status dnsmasq
● dnsmasq.service - DNS caching server.
   Loaded: loaded (/usr/lib/systemd/system/dnsmasq.service; enabled; vendor preset: disabled)
   Active: active (running) since Mon 2017-08-28 17:35:34 EDT; 25s ago
 Main PID: 1113 (dnsmasq)
   Memory: 608.0K
   CGroup: /system.slice/dnsmasq.service
           └─1113 /usr/sbin/dnsmasq -k

Aug 28 17:35:35 master.ocpcluster.local dnsmasq[1113]: using nameserver 192.168.20.1#53 for domain ocpcluster.local
Aug 28 17:35:35 master.ocpcluster.local dnsmasq[1113]: read /etc/hosts - 2 addresses
Aug 28 17:35:49 master.ocpcluster.local dnsmasq[1113]: setting upstream servers from DBus
**Aug 28 17:35:49 master.ocpcluster.local dnsmasq[1113]: using nameserver 127.0.0.1#53 for domain cluster.local**
Aug 28 17:35:49 master.ocpcluster.local dnsmasq[1113]: using nameserver 127.0.0.1#53 for domain in-addr.arpa
Aug 28 17:35:49 master.ocpcluster.local dnsmasq[1113]: using nameserver 192.168.20.1#53
Aug 28 17:35:49 master.ocpcluster.local dnsmasq[1113]: using nameserver 8.8.8.8#53
Aug 28 17:35:49 master.ocpcluster.local dnsmasq[1113]: using nameserver 192.168.20.1#53 for domain ocpcluster.local
Aug 28 17:35:49 master.ocpcluster.local dnsmasq[1113]: using nameserver 127.0.0.1#53 for domain in-
```

env: libvirt/kvm

**Comment by mangirdaz** [comment]
we add all required details to containers:
```
bash-4.2$ cat /etc/resolv.conf                                                                                                                                                                                    
nameserver 192.168.20.160
search kube-service-catalog.svc.cluster.local svc.cluster.local cluster.local ocpcluster.local
options ndots:5
```

but not on the nodes/masters. and ansible runs on master to check this

**Comment by juanluisvaladas** [comment]
@sdodson I don't think the cluster.local. has ever been added by the dispatcher.

I can confirm right now OCP (not origin) doesn't add it in at least 3.4, 3.5 and 3.6

**Comment by mangirdaz** [comment]
@djwhatle Ping ^

**Comment by sdodson** [comment]
@juanluisvaladas it should happen here https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_node_dnsmasq/files/networkmanager/99-origin-dns.sh#L115-L117

Can you get me `journalctl --no-pager -u NetworkManager-dispatcher` ?

**Comment by mangirdaz** [comment]
I think it does not do whats expected because all cluster I use has this as domain `appcluster.local` , `ocpcluster.local`, etc.... so my lab is replica for this... Changes code in pull request to fix this in dispatcher.. 
```
Sep 05 13:39:49 master.ocpcluster.local nm-dispatcher[97934]: + '[' 0 -eq 1 ']'
Sep 05 13:39:49 master.ocpcluster.local nm-dispatcher[97934]: ++ systemctl -q is-active dnsmasq.service
Sep 05 13:39:49 master.ocpcluster.local nm-dispatcher[97934]: + grep -q 99-origin-dns.sh /etc/resolv.conf
Sep 05 13:39:49 master.ocpcluster.local nm-dispatcher[97934]: + sed -e '/^nameserver.*$/d' /etc/resolv.conf
Sep 05 13:39:49 master.ocpcluster.local nm-dispatcher[97934]: + echo 'nameserver 192.168.20.160'
Sep 05 13:39:49 master.ocpcluster.local nm-dispatcher[97934]: + grep -q 'search.*cluster.local' /tmp/tmp.VnbbEpWILX
Sep 05 13:39:49 master.ocpcluster.local nm-dispatcher[97934]: + cp -Z /tmp/tmp.VnbbEpWILX /etc/resolv.conf
Sep 05 13:39:49 master.ocpcluster.local nm-dispatcher[97934]: + rm -f /tmp/tmp.OpPhhX8tl9 /tmp/tmp.F7wZrRgTvM /tmp/tmp.mTvjs8skAE /tmp/tmp.VnbbEpWILX
```
And because serach matches it fails...

**Comment by mangirdaz** [comment]
Should we add ndot:5 too?

**Comment by sdodson** [comment]
> Should we add ndot:5 too?

That seems like a pretty drastic change to make at the host level. It'd force all hostname lookups with less than 5 dots to go through the search path.

Looking through the docs for `man resolv.conf` it makes it sounds as if the default is 1, which means that it will first attempt to look up `docker-registry.default.svc` which will fail and then it'll start walking through the search paths. I guess this introduces the risk that if that domain name existed we'd attempt to push/pull an image from there but `.svc` isn't a valid TLD.

**Comment by mangirdaz** [comment]
Make sense. Will test it more on my clusters and let you know.Somehow missed this dispatcher change :)

File: roles/openshift_node_dnsmasq/files/networkmanager/99-origin-dns.sh [hunk 1]
```diff
@@ -112,7 +112,7 @@ EOF
       fi
       sed -e '/^nameserver.*$/d' /etc/resolv.conf >> ${NEW_RESOLV_CONF}
       echo "nameserver "${def_route_ip}"" >> ${NEW_RESOLV_CONF}
-      if ! grep -q 'search.*cluster.local' ${NEW_RESOLV_CONF}; then
+      if ! grep -q 'search.*/scluster.local' ${NEW_RESOLV_CONF}; then
```
Comment (sdodson) [inline roles/openshift_node_dnsmasq/files/networkmanager/99-origin-dns.sh]: I think we should template this file now so we can use `{{ openshift.common.dns_domain }}`

Comment (mangirdaz) [reply roles/openshift_node_dnsmasq/files/networkmanager/99-origin-dns.sh]: to move dispatcher script to the template and use ansible to manage this file script?

Comment (sdodson) [reply roles/openshift_node_dnsmasq/files/networkmanager/99-origin-dns.sh]: Yeah, so that we can check for the proper dns zone if it's been reconfigured.

Comment (mangirdaz) [reply roles/openshift_node_dnsmasq/files/networkmanager/99-origin-dns.sh]: lets merge this as hotfix and I can do this sometimes in next few weeks when I have time if this is not in rush. We can track it in the issues or trello :)

**Pull request closed without merging**